### PR TITLE
Automatically sort the input sequence in links.loss.CRF1d

### PR DESCRIPTION
**Pull request by himkt**
- [x] flake8
- [x] compatibility
- [x] test

CRF1d needs to sort the input sequences
in descending order of lengths and transpose the sequences.
But in NStepLSTM, this procedure is not needed
because links.connection.NStepRNNBase already does.

It would be better that CRF1d also does this, which allows users
to concentrate on defining the network architecture.

So I changed the links.CRF1d to sort the input sequence automatically and send the PR.
After deal with the API compatibility policy, I'll remove WIP prefix.

### CODE REVIEW

**Comment by himkt** [comment]
I checked and updated the links.loss.CRF1d.argmax to return a list of Variable.
In this modification, this PR seems not to break compatibility.
So I removed WIP prefix, please check it.

There are two remaining which I can do.

1. There is no test for links.loss.CRF1d (if it is needed, I'll create some tests)
2. NStepRNN and CRF1d use the same `permutate_list` and `argsort_list_descent`
- I think it would be better to unify but I'm not sure where module is the best for them.

cc: @aonotas @unnonouno

**Comment by unnonouno** [comment]
Check comments.

**Comment by himkt** [comment]
@unnonouno I'm very sorry but I could not find what I should do (I overlooked something important?).
Could you please tell me what I can check?

**Comment by himkt** [comment]
Thanks, I updated the PR.
BTW, what the situation does it break compatibility?
(I thought sorting and reindexing is totally done in `forward` and `argmax`, which keep the compatibility)

**Comment by unnonouno** [comment]
Sorry for late reply. If the default value of `transpose` argument is `True`, programs which use the old interface will be broken.

**Comment by unnonouno** [comment]
Can you write a test using `transpose=True`?

**Comment by himkt** [comment]
@unnonouno 
Sorry for late reply...
I'm not sure what I should test ...
I'm sorry but could you please tell me what I should check or please take over my PR:bow:
(If you will take over my work, I'll close this PR:pray:)

By the way, I previously added `transpose` option to `forward` and `argmax`.
But now, I think it would be better to add `transpose` option to `__init__`.
Please please tell me what you thinks about it.

**Comment by himkt** [comment]
Now, I will make two changes to this PR.
- Remove `transpose` option from `__call__` and `argmax` and add it to `__init__`
- Add test to check whether if the output from `__call__` in `transpose=False` and it in `transpose=True` are the same

File: chainer/links/loss/crf1d.py [hunk 1]
```diff
@@ -25,7 +42,13 @@ def __init__(self, n_label):
             self.cost = variable.Parameter(0, (n_label, n_label))
 
     def forward(self, xs, ys, reduce='mean'):
```
[the remark this answers is not in the document]
Comment (himkt) [reply chainer/links/loss/crf1d.py]: I updated my PR to set `transpose` in `__init__` for compatibility.

**Comment by himkt** [comment]
@unnonouno 
Thanks for your advice, I updated my PR.

> And, OK i'll take over this PR and write a unit test. Please keep this PR open. I'll send another one.
I got it, I keep this PR open :bow:

**Comment by himkt** [comment]
@unnonouno What is the status of this PR?
I'm sorry to leave it to you all.
If you are busy, please let me know how I should write tests :bow:

**Comment by himkt** [comment]
Now this PR should be merged after #5326

**Comment by niboshi** [comment]
@unnonouno Could you check again?

**Comment by niboshi** [comment]
@unnonouno ping

**Comment by himkt** [comment]
https://github.com/chainer/chainer/pull/6351 took over this PR.
So, I close this PR, thanks a lot, @unnonouno :bow:

**Pull request closed without merging**